Data/Row/Variants.hs

```diff
@@ -408,6 +411,112 @@ fromLabels mk = getCompose $ metamorph @_ @ρ @c @FlipConst @(Const ()) @(Compos
         doCons l (FlipConst (Compose v)) = Compose $ IsJust l <$> mk l <|> extend @τ l <$> v
           \\ extendHas @ρ @ℓ @τ
 
+{--------------------------------------------------------------------
+  Functions for variants of ApSingle
+--------------------------------------------------------------------}
+
+newtype VApS x (fs :: Row (* -> *)) = VApS { unVApS :: Var (ApSingle fs x) }
+newtype FlipApp (x :: *) (f :: * -> *) = FlipApp (f x)
+
+eraseSingle
+  :: forall (c :: (* -> *) -> Constraint) (fs :: Row (* -> *)) (x :: *) (y :: *)
+   . (Forall fs c)
+  => (forall f . (c f) => f x -> y)
+  -> Var (ApSingle fs x)
+  -> y
+eraseSingle f =
+  getConst
+    . metamorph @_ @fs @c @Either @(VApS x) @(Const y) @(FlipApp x)
+        Proxy
+        doNil
+        doUncons
+        doCons
+    . VApS
+ where
+  doNil = impossible . unsafeCoerce . unVApS
```

Comment:
I'm not sure why you should need `unsafeCoerce` here.  After you unwrap the `VApS`, you should have a value of type `Var (ApSingle Empty x)`, and I would think that type family resolution would turn this into `Var Empty`, which means you can use `impossible` directly.  Is this not the case?  What error message do you get?

Reply:
Taking out `unsafeCoerce` works fine. I must have thought it was required when I was working through other type errors. It will be removed in the next push I make.